contracts/AgentMech.sol

```diff
@@ -165,44 +203,22 @@ contract AgentMech is ERC721Mech {
 
         // Delete the delivered element from the map
         delete mapRequestIds[requestId];
-        delete mapRequestAddresses[requestId];
-    }
-
-    /// @dev Changes mech marketplace address.
-    /// @param newMechMarketplace New mech marketplace address.
-    function changeMechMarketplace(address newMechMarketplace) external onlyOperator {
```

Comment:
Not needed anymore, we stick to the specified marketplace, otherwise things become super convoluted. If there is a need to change the marketplace, makes sense to create another mech.